### PR TITLE
Update primitives information in the Preview CLI reference

### PR DESCRIPTION
**Pull request by keerlu**
Add `migrate diff` and `db execute` information from CLI help, and reformat to be in the same style as the non-Preview command reference.

### CODE REVIEW

File: content/200-concepts/100-components/250-preview-features/080-cli-preview-features.mdx [hunk 1]
```diff
@@ -9,40 +10,168 @@ To use a Preview CLI feature, [use the `--preview-feature` flag](./) when you ru
...
-This command applies a SQL script to the database without interacting with the Prisma migrations table. The script can be passed via a file path or stdin.
+One of the following `--from-...` options is required:
 
-The format of the command is:
+| Options                   | Description                                     |
+| :------------------------ | :---------------------------------------------- |
+| `--from-url`              | A data source URL                               |
+| `--from-migrations`       | Path to the Prisma Migrate migrations directory |
+| `--from-schema-datamodel` | Path to an empty schema file                    |
+| `--from-empty`            | An empty data source                            |
```
Comment (Jolg42) [inline content/200-concepts/100-components/250-preview-features/080-cli-preview-features.mdx]:
```suggestion
| Options                   | Description                                     |
| :------------------------ | :---------------------------------------------- |
| `--from-url`              | A data source URL                               |
| `--from-migrations`       | Path to the Prisma Migrate migrations directory |
| `--from-schema-datamodel` | Path to a Prisma schema file, uses the datamodel for the diff |
| `--from-schema-datasource` | Path to a Prisma schema file, uses the datasource url for the diff |
| `--from-empty`            | Flag to assume from is an empty datamodel |
```

File: content/200-concepts/100-components/250-preview-features/080-cli-preview-features.mdx [hunk 2]
```diff
@@ -9,40 +10,168 @@ To use a Preview CLI feature, [use the `--preview-feature` flag](./) when you ru
...
-```
+One of the following `--to-...` options is required:
 
-For details of the available flags for this command, run:
+| Options                 | Description                                     |
+| :---------------------- | :---------------------------------------------- |
+| `--to-url`              | A data source URL                               |
+| `--to-migrations`       | Path to the Prisma Migrate migrations directory |
+| `--to-schema-datamodel` | Path to an empty schema file                    |
+| `--to-empty`            | An empty data source                            |
```
Comment (Jolg42) [inline content/200-concepts/100-components/250-preview-features/080-cli-preview-features.mdx]:
```suggestion
| Options                 | Description                                     |
| :---------------------- | :---------------------------------------------- |
| `--to-url`              | A data source URL                               |
| `--to-migrations`       | Path to the Prisma Migrate migrations directory |
| `--to-schema-datamodel` | Path to a Prisma schema file, uses the datamodel for the diff |
| `--to-schema-datasource` | Path to a Prisma schema file, uses the datasource url for the diff |
| `--to-empty`             | Flag to assume to is an empty datamodel             |
```

File: content/200-concepts/100-components/250-preview-features/080-cli-preview-features.mdx [hunk 3]
```diff
@@ -9,40 +10,168 @@ To use a Preview CLI feature, [use the `--preview-feature` flag](./) when you ru
...
 
-```terminal
-npx prisma db execute --help
+Other options:
+
+| Options                | Required | Description                                                         | Notes                                                           |
+| :--------------------- | :------- | :------------------------------------------------------------------ | :-------------------------------------------------------------- |
+| `-shadow-database-url` | No       | URL for the shadow database                                         | Only required if using `--to-migrations` or `--from-migrations` |
```
Comment (Jolg42) [inline content/200-concepts/100-components/250-preview-features/080-cli-preview-features.mdx]:
```suggestion
| `--shadow-database-url` | No       | URL for the shadow database                                         | Only required if using `--to-migrations` or `--from-migrations` |
```

File: content/200-concepts/100-components/250-preview-features/080-cli-preview-features.mdx [hunk 4]
```diff
@@ -9,40 +10,168 @@ To use a Preview CLI feature, [use the `--preview-feature` flag](./) when you ru
...
+    --to-url "postgresql://login:password@localhost:5432/db2"
+  ```
+
+- Compare the state of a database with a URL of `$DATABASE_URL` to the schema defined by the migrations in the `./migrations` directory, and output the differences to a script `script.sql`:
+
+  ```terminal
+  prisma migrate diff \
+   --preview-feature \
+   --from-url "$DATABASE_URL" \
+   --to-migrations ./migrations --script > script.sql
```
Comment (Jolg42) [inline content/200-concepts/100-components/250-preview-features/080-cli-preview-features.mdx]:
```suggestion
   --to-migrations ./migrations \
   --script > script.sql
```

File: content/200-concepts/100-components/250-preview-features/080-cli-preview-features.mdx [hunk 5]
```diff
@@ -9,40 +10,168 @@ To use a Preview CLI feature, [use the `--preview-feature` flag](./) when you ru
...
+
+The output of the command is connector-specific, and is not meant for returning data, but only to report success or failure.
+
+See also:
+
+- [Migration troubleshooting in production](/guides/database/production-troubleshooting#fixing-failed-migrations-with-migrate-diff-and-db-execute) <span class="guide"></span>
+
+#### Prerequisites
+
+Before using the `db execute` command, you must define a valid [`datasource`](/concepts/components/prisma-schema/data-sources) within your `schema.prisma` file.
```
Comment (Jolg42) [inline content/200-concepts/100-components/250-preview-features/080-cli-preview-features.mdx]: Not a prerequisite if providing the url via `--url=...`

Comment (keerlu) [reply content/200-concepts/100-components/250-preview-features/080-cli-preview-features.mdx]: Thanks, have clarified this

**Comment by Jolg42** [comment]
Preview at https://deploy-preview-2837--prisma2-docs.netlify.app/docs/concepts/components/preview-features/cli-preview-features

File: content/200-concepts/100-components/250-preview-features/080-cli-preview-features.mdx [hunk 6]
```diff
@@ -9,40 +10,171 @@ To use a Preview CLI feature, [use the `--preview-feature` flag](./) when you ru
...
+See also:
 
-For details of the available flags for this command, run:
+- [Migration troubleshooting in production](/guides/database/production-troubleshooting#fixing-failed-migrations-with-migrate-diff-and-db-execute) <span class="guide"></span>
 
-```terminal
-npx prisma migrate diff --help
+#### Prerequisites
+
+Before using the `migrate diff` command, you must define a valid [`datasource`](/concepts/components/prisma-schema/data-sources) within your `schema.prisma` file.
```
Comment (Jolg42) [inline content/200-concepts/100-components/250-preview-features/080-cli-preview-features.mdx]: It's not a prerequisites as you can use from/to
- --from-url
- --from-migrations
- --from-schema-datamodel
- --from-empty`   

only --from-schema-datasource (or to) requires a valid datasource

File: content/200-concepts/100-components/250-preview-features/080-cli-preview-features.mdx [hunk 7]
```diff
@@ -9,40 +10,171 @@ To use a Preview CLI feature, [use the `--preview-feature` flag](./) when you ru
...
-This command applies a SQL script to the database without interacting with the Prisma migrations table. The script can be passed via a file path or stdin.
+One of the following `--from-...` options is required:
 
-The format of the command is:
+| Options                    | Description                                                          |
+| :------------------------- | :------------------------------------------------------------------- |
+| `--from-url`               | A data source URL                                                    |
+| `--from-migrations`        | Path to the Prisma Migrate migrations directory                      |
+| `--from-schema-datamodel`  | Path to a Prisma schema file, uses the data model for the diff       |
+| `--from-schema-datasource` | Path to a Prisma schema file, uses the `datasource` URl for the diff |
```
Comment (Jolg42) [inline content/200-concepts/100-components/250-preview-features/080-cli-preview-features.mdx]:
```suggestion
| `--from-schema-datasource` | Path to a Prisma schema file, uses the `datasource` URL for the diff |
```

File: content/200-concepts/100-components/250-preview-features/080-cli-preview-features.mdx [hunk 8]
```diff
@@ -9,40 +10,171 @@ To use a Preview CLI feature, [use the `--preview-feature` flag](./) when you ru
...
 
-```terminal
-npx prisma db execute --preview-feature --url $PROD_DATABASE --file script.sql
-```
+One of the following `--to-...` options is required:
 
-For details of the available flags for this command, run:
+| Options                  | Description                                                          |
+| :----------------------- | :------------------------------------------------------------------- |
+| `--to-url`               | A data source URL                                                    |
```
Comment (do4gr) [inline content/200-concepts/100-components/250-preview-features/080-cli-preview-features.mdx]: Everywhere else we use `datasource` as one word

**Comment by do4gr** [comment]
Looks great. I think there is one more thing we should add to the diff explanation: diff only operates on things Prisma supports in the PSL already (that means no views, triggers etc..). If two databases only differ in such entities diff will not show a difference. It will also not produce SQL to create those in that case. I think that is sth people need to be aware of. I'd probably place this up front as an admonition.

File: content/200-concepts/100-components/250-preview-features/080-cli-preview-features.mdx (hunk 8 above)
Comment (keerlu) [reply content/200-concepts/100-components/250-preview-features/080-cli-preview-features.mdx]: Hm, my understanding from searching the docs is that we use `datasource` when directly referring to the code block, and 'data source' for the generic concept. E.g. [here](https://www.prisma.io/docs/concepts/components/prisma-schema/data-sources): "A data source determines how Prisma connects your database, and is represented by the `datasource`  block in the Prisma schema."

Unless this is actually supposed to be referring to the `datasource` block, in which case it needs backticks around it?

Comment (keerlu) [reply content/200-concepts/100-components/250-preview-features/080-cli-preview-features.mdx]: Oh I see, everywhere else in this document... yeah, agree, it's a bit muddled. I'll update them all to be consistent.

File: content/200-concepts/100-components/250-preview-features/080-cli-preview-features.mdx (hunk 6 above)
Comment (keerlu) [reply content/200-concepts/100-components/250-preview-features/080-cli-preview-features.mdx]: Thanks, done

File: content/200-concepts/100-components/250-preview-features/080-cli-preview-features.mdx (hunk 8 above)
Comment (keerlu) [reply content/200-concepts/100-components/250-preview-features/080-cli-preview-features.mdx]: Hopefully that's clearer now

**Comment by keerlu** [comment]
> Looks great. I think there is one more thing we should add to the diff explanation: diff only operates on things Prisma supports in the PSL already (that means no views, triggers etc..). If two databases only differ in such entities diff will not show a difference. It will also not produce SQL to create those in that case. I think that is sth people need to be aware of. I'd probably place this up front as an admonition.

Thanks, I've added an admonition. Would be nice to have a link for the Prisma Schema Language but I'm actually not sure what to link to... would the schema reference work?

**Pull request merged**